src/rank_llm/result.py

```diff
@@ -0,0 +1,59 @@
+import json
+from typing import List, Dict, Any
+
+
+class RankingExecInfo:
+    def __init__(
+        self, prompt, response: str, input_token_count: int, output_token_count: int
+    ):
+        self.prompt = prompt
+        self.response = response
+        self.input_token_count = input_token_count
+        self.output_token_count = output_token_count
+
+
+class Result:
+    def __init__(
+        self,
+        query: str,
+        hits: List[Dict[str, Any]],
+        ranking_exec_summary: List[RankingExecInfo] = None,
+    ):
+        self.query = query
+        self.hits = hits
+        self.ranking_exec_summary = ranking_exec_summary
+
+    def __repr__(self):
+        return str(self.__dict__)
+
+
+class ResultsWriter:
+    def __init__(self, results: List[Result], append: bool = False):
+        self._results = results
+        self._append = append
+
+    def write_ranking_exec_summary(self, filename: str):
+        exec_summary = {}
+        for result in self._results:
+            key = result.query
+            values = []
+            for info in result.ranking_exec_summary:
+                values.append(info.__dict__)
+            exec_summary[key] = values
```

Comment:
No it would not since the query is the key, but this is identical to the current behavior in input token file where query is the key:

```
{
    "How much impact do masks have on preventing the spread of the COVID-19?": [
        35947,
        810
    ],
    ...
}
```
I can change this to an array of dictionaries instead, with "query" and "ranking_exec_summary" as keys, WDYT?

Reply:
done

Reply:
yep that array idea works better! thanks